frontend/src/common/queries/collections.ts

```diff
@@ -258,15 +330,16 @@ const editCollection = async function ({
   return result;
 };
 
-export function useEditCollection() {
+export function useEditCollection(collectionID: Collection["id"]) {
   const queryCache = useQueryCache();
 
   return useMutation(editCollection, {
-    onSuccess: (collection: Collection) => {
-      return queryCache.setQueryData(
-        [USE_COLLECTION, collection.id, collection.visibility],
-        collection
-      );
+    onSuccess: () => {
+      queryCache.invalidateQueries([
+        USE_COLLECTION,
+        collectionID,
+        VISIBILITY_TYPE.PRIVATE,
+      ]);
```

Comment:
instead of optimistically updating the collection, just invalidate the query to recalculate the revision attributes